lib/node_modules/@stdlib/lapack/base/dlartv/lib/base.js

```diff
@@ -0,0 +1,72 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-params */
+
+'use strict';
+
+// MAIN //
+
+/**
+* Applies a vector of real plane rotations to elements of real vectors `X` and `Y`.
+*
+* @private
+* @param {NonNegativeInteger} N - number of plane rotations to apply
+* @param {Float64Array} X - input vector
+* @param {integer} strideX - stride length for `X`
+* @param {NonNegativeInteger} offsetX - starting index of `X`
+* @param {Float64Array} Y - input vector
+* @param {integer} strideY - stride length for `Y`
+* @param {NonNegativeInteger} offsetY - starting index of `Y`
+* @param {Float64Array} C - vector of cosines of plane rotations
+* @param {integer} strideC - stride length for `C`
+* @param {NonNegativeInteger} offsetC - starting index of `C`
+* @param {Float64Array} S - vector of sines of plane rotations
+* @param {integer} strideS - stride length for `S`
+* @param {NonNegativeInteger} offsetS - starting index of `S`
+* @returns {Float64Array} `X`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var X = new Float64Array( [ 1.0, 1.0 ] );
+* var Y = new Float64Array( [ 1.0, 1.0 ] );
+* var C = new Float64Array( [ 1.0, 1.0 ] );
+* var S = new Float64Array( [ 0.0, 0.0 ] );
+*
+* X = dlartv( 2, X, 1, 0, Y, 1, 0, C, 1, 0, S, 1, 0 );
+* // X => <Float64Array>[ 1.0, 1.0 ]
+*/
+function dlartv( N, X, strideX, offsetX, Y, strideY, offsetY, C, strideC, offsetC, S, strideS, offsetS ) {
+	var xi;
+	var yi;
+	var i;
+
+	for ( i = 0; i < N; i++ ) {
+		xi = X[ offsetX + ( i * strideX ) ];
```

Comment:
@Pranavchiku Please refactor to reduce flops. This is a common occurrence in your PRs. E.g.,

```javascript
iy = offsetY;
for ( i = 0; i < N; i++ ) {
	yi = Y[ iy ];
    
	// ...

	iy += strideY;
}
```

Notice how for every pointer computation, you have 2 flops (1 multiplication and 1 addition). Notice in the example above, there is a single flop.

Reply:
While temporary variables are not without cost, they will certainly help improve readability in the logic which follows.